fail2ban/server/filter.py

```diff
@@ -844,35 +943,50 @@ def getJournalMatch(self, match): # pragma: no cover - Base class, not used
 
 import socket
 import struct
+from .utils import Utils
 
 
 class DNSUtils:
 
 	IP_CRE = re.compile("^(?:\d{1,3}\.){3}\d{1,3}$")
 
+	# todo: make configurable the expired time and max count of cache entries:
+	CACHE_dnsToIp = Utils.Cache(maxCount=1000, maxTime=60*60)
```

Comment:
Why should fail2ban get involved in such a low-level operation? To save maybe 150ms per lookup (how often are you logging DNS names?), I'm just asking...what are we really looking at here....let's be realistic.